enhancements/update/distribute-secondary-metadata-as-container-image.md

```diff
@@ -0,0 +1,197 @@
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [x] Graduation criteria
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions
+
+### Content trust
+Trust in the container image registry and in the container image content is possibly an issue if we follow an approach where Cincinnati uses a mutable tag, e.g. `latest` tag to receive the most recent content from the container registry.
+Assuming we're using Quay as an image registry, [which does not support Docker Content Trust](https://support.coreos.com/hc/en-us/articles/115000289114-Docker-Content-Trust-support).
+If we decide to ensure image integrity, which means do we have that do not further complicate the workflow?
+* An option is to package the signatures as a container image as well, and publish them in a separate repository, or in the same repository under a well-known tag e.g. `latest-signature`.
+  This option is currently described in this enhancement.
+
+### Product name for the CPaaS integration
+Which product name should this image be released under?
+Related to this is that it's currently unclear if the Cincinnati image itself will be published under its own product name or now.
+Further note: the repository for the graph-data images will not be semantically versioned.
+
+### Repository location and name
+Under which well-known container image repository will we publish the images?
+
+### Continuous build and release
+Can CPaaS handle continuous build and release from master?
+
+## Summary
+The [cincinnati-graph-data GitHub repository][cincinnati-graph-data] will be continuously packaged and released in form of a container image by the means of integration with CPaaS.
+This image will be available for public consumption, and will be used by the publicly available Cincinnati instances operated by Red Hat.
+The image can be transferred and served to Cincinnati instances operated by any party, including but not limited to network-restricted environments.
+The source of the container image, namely the cincinnati-graph-data, will remain publicly available and its workflows unaffected by this enhancement.
+
+## Motivation
+
+### Goals
+1. The container image will become the primary way of consuming and distributing the content of the cincinnati-graph-data repository.
+2. As long as the primary release metadata will be packaged, distributed and consumed in form of container images as well, the transport mechanism required for data consumption by Cincinnati, as well as mirroring and serving its consumed content, will be solely [Docker Registry HTTP API V2][docker-registry-http-api-v2] protocol.
+
+
+### Non-Goals
+1. Changes to the GitHub repository workflow or data format.
+2. Implementation details about the streamlining of the mirroring process for primary and secondary release metadata.
```

Comment:
The `oc` command currently has features to mirror OCP releases and to mirror optional operators. It seems natural for it to also mirror these graph data images. Would that be a separate enhancement proposal? Or why not part of this one?

Reply:
I purposely left this detail out of the proposal to increase to keep the scope as small as possible. I do see this as a possible improvement in the future, just not critical to this enhancement.

Reply:
Ok. We just need to ensure that there is some viable and supportable way for the customer to accomplish the mirroring. Perhaps `skopeo` would meet this need, but it's worth confirming and testing.

Reply:
I agree. I'm not worried about this though, as anything which can pull an image to disk and push it from disk will do. [skopeo seems do too](https://github.com/containers/skopeo#copying-images) as well. But I think that some subcommand of `oc` can also copy images.

Reply:
Do we need a ticket somewhere to track this? I wouldn't really know where that would be. Could it be an issue in this repository?

Reply:
We need to create a Jira card track this (to create an enhancement PR in future) or create another enhancement PR around this.